Fixes #38591 - Api calls refactor

Lukshio opened the pull request:
Remove useApi duplicates, unify calls

ofedoren commented on webpack/JobInvocationDetail/JobInvocationHostTable.js:
```diff
@@ -47,25 +42,53 @@ const JobInvocationHostTable = ({
   const history = useHistory();
+  const dispatch = useDispatch();
+
+  const [APIResponse, setAPIResponse] = useState([]);
```
quick nit: doesn't `APIResponse` means it's a constant now? Although JS is weird and it might work just fine, but I'd rather change it to `apiResponse` or similar.

Lukshio replied:
Yep, it will make more sense with camelCase, thanks.

adamruzicka commented:
Needs a rebase

Lukshio commented:
> This makes it so if a users selects all hosts, and opens in a new tab, only the first page is opened, instead of the first 100

Can you please retest it now? Opening in new tabs has its own limit https://github.com/theforeman/foreman_remote_execution/blob/2310a7388226b71584538f58cf414d84532f67c4/webpack/JobInvocationDetail/CheckboxesActions.js#L83-L86

MariaAga commented:
since nothing calls for the first 100 ( per_page = MAX_HOSTS_API_SIZE) it still only opens the first page instead of the first 100

MariaAga commented:
also it auto sets per page to 20 instead of the user per_page

MariaAga commented:
changing per_page adds brackets around the search, and then every page change adds more brackets, searching by clicking on the chart doesnt work anymore

Lukshio commented:
> changing per_page adds brackets around the search, and then every page change adds more brackets, searching by clicking on the chart doesnt work anymore
Fixed


> also it auto sets per page to 20 instead of the user per_page
Fixed



> since nothing calls for the first 100 ( per_page = MAX_HOSTS_API_SIZE) it still only opens the first page instead of the first 100

I couldn't replicate this issue

MariaAga commented:
to replicate - have a job run on 21+ hosts, open the job details page, make sure per page is less than the amount of hosts (usually set to 20) click on the open all tab, the number shown in the modal is the per page number, and not all hosts number

Lukshio commented:
> to replicate - have a job run on 21+ hosts, open the job details page, make sure per page is less than the amount of hosts (usually set to 20) click on the open all tab, the number shown in the modal is the per page number, and not all hosts number

Added old MAX_HOSTS const to load only first 100

adamruzicka commented:
It seems we're polling periodically, without waiting for the previous poll to finish? Could we do something about that?

<img width="2164" height="302" alt="image" src="https://github.com/user-attachments/assets/e8f32d7c-daa0-47f6-94bd-6ddc459ef7af" />

Lukshio commented:
After discussion with  @MariaAga new behavior is:
- Open in new tabs in applied to current page if not selected, same for failed open
- All api calls are reduced to 1 which handles ids for selection, failed, etc
- MAX_HOSTS is no longer in use

adamruzicka commented:
Needs a rebase now

Lukshio commented:
> Needs a rebase now

Rebased

Lukshio commented:
> > The open in new tab button - should open the page if nothing is selected, should not have the text "The maximum is 100.«", and should have a matching text (in the modal and tooltip) if its opening the selected in the page or the just the page. open all failed - should say open all failed runs from this page, the modal also shouldnt say "the maximum is 100.«", also ,it doesnt open any tabs when clicked
> 
> Unable to reproduce not opening page if nothing selected, other fixed

Ad. refactored opening new tabs, now it should be working

MariaAga commented on webpack/JobInvocationDetail/JobInvocationHostTable.js:
```diff
@@ -26,52 +27,71 @@ import PropTypes from 'prop-types';
+    () => ({
+      page: urlPage ? Number(urlPage) : 1,
+      ...(urlPerPage && { per_page: Number(urlPerPage) }),
+      order: urlOrder || '',
```
If I change the order to name desc, and the change the page the order gets ignored. Also it doesnt seem to go to the url which might be the reason

Lukshio commented:
> clicking the chart seems to add conditions instead of replacing them `(job_invocation.result = failed) AND ((job_invocation.result = success) AND ((job_invocation.result = failed) AND ((job_invocation.result = cancelled) AND ((job_invocation.result = pending) AND ((job_invocation.result = success) AND ((job_invocation.result = failed)))))))`

re fixed it, hopefully this will stop the circle of fixing something, and breaking something else :)

MariaAga reviewed (commented):
Seems to work as expected now! thats quite a lot of changes, @kmalyjur can you take a look as well to make sure I'm not missing anything?

kmalyjur commented on webpack/JobInvocationDetail/OpenAllInvocationsModal.js:
```diff
@@ -25,21 +21,35 @@ export const PopupAlert = ({ setShowAlert }) => (
+        </>
+      );
+    }
+    return <>{__('Opening current page')} </>;
```
I might be nitpicking, but isn't that line unnecessary? It's already mentioned twice in the modal that it's the current page. The number of hosts to be opened, as for the rest of the modal possibilities, might be more useful. 🤔

kmalyjur commented on webpack/JobInvocationDetail/OpenAllInvocationsModal.js:
```diff
@@ -25,21 +21,35 @@ export const PopupAlert = ({ setShowAlert }) => (
   isOpenFailed,
-  jobID,
   onClose,
-  setShowAlert,
```
Delete `jobID` and `setShowAlert` from CheckboxesActions.js, too

Lukshio replied:
Maybe I am misunderstanding, but these two are still in use in CheckboxesActions.js

Lukshio replied:
I understand now, thank you, fixed.

kmalyjur commented on webpack/JobInvocationDetail/JobInvocationHostTable.js:
```diff
@@ -351,21 +341,21 @@ const JobInvocationHostTable = ({
           params={{
-            page: response?.page || Number(urlPage),
-            per_page: response?.per_page || Number(urlPerPage),
+            page: defaultParams.page || Number(urlPage),
```
Missing `?` (although I think it's not needed, it would be inconsistent)

kmalyjur commented on webpack/JobInvocationDetail/JobInvocationHostTable.js:
```diff
@@ -26,52 +27,76 @@ import PropTypes from 'prop-types';
+import { PopupAlert } from './OpenAllInvocationsModal';
 
 const JobInvocationHostTable = ({
-  failedCount,
```
Needs to be deleted from index.js, too

kmalyjur reviewed (changes requested):
Thank you, it works as it should, and the code is great except for these minor things.

Lukshio commented on webpack/JobInvocationDetail/CheckboxesActions.js:
```diff
@@ -282,16 +293,14 @@ export const CheckboxesActions = ({
         onClose={() => setIsModalOpen(false)}
         failedCount={failedCount}
-        failedHosts={failedHosts}
         jobID={jobID}
```
Sure, Thanks.
